trl/trainer/sft_trainer.py

```diff
@@ -469,10 +496,14 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
         if mode == "train":
             # When using padding-free, the attention_mask is not present in the inputs, instead we have cu_seq_lens_q,
             # cu_seq_lens_k, and max_length_k, max_length_q and position_ids.
-            if "cu_seq_lens_q" in inputs:
-                num_tokens_in_batch = self.accelerator.gather_for_metrics(inputs["cu_seq_lens_q"][-1]).sum().item()
-            else:
+            if "attention_mask" in inputs:
                 num_tokens_in_batch = self.accelerator.gather_for_metrics(inputs["attention_mask"].sum()).sum().item()
+            elif "position_ids" in inputs:
+                num_tokens_in_batch = (
+                    self.accelerator.gather_for_metrics(torch.tensor(inputs["position_ids"].size(1))).sum().item()
```

Comment:
```suggestion
                    self.accelerator.gather_for_metrics(torch.tensor(inputs["position_ids"].size(1)).sum()).sum().item()
```

Reply:
are you sure you need it? `size(1)` returns a single int